src/serialization/sb2.js

```diff
@@ -1070,8 +1079,22 @@ const parseBlock = function (sb2block, addBroadcastMsg, getVariableId, extension
             argumentids: JSON.stringify(parseProcedureArgIds(sb2block[1]))
         };
     } else if (oldOpcode === 'getParam') {
+        let returnCode = sb2block[2];
+
+        // Ensure the returnCode is "b" if used in a boolean input.
+        if (parentExpectedArg && (
+            // Used as a CONDITION input like for control_if.
+            parentExpectedArg.inputName === 'CONDITION' ||
```

Comment:
this check seems a little brittle, and it doesn't cover all our boolean cases (for example the not operator input is named OPERAND). The name doesn't actually prescribe the type information. Maybe we should add an additional property in the (sb2_specmap)[https://github.com/LLK/scratch-vm/blob/develop/src/serialization/sb2_specmap.js] for all the boolean inputs we know about? That seems like an approach that is more consistent with everything else we're doing for sb2 import.

Reply:
hmmm or actually....

I wonder if you can just get away with the following check `parentExpectedArg.type === 'input' && !parentExpectedArg.inputOp`